src/components/ApiMenu/ApiMenu.tsx

```diff
@@ -40,7 +43,7 @@ const ApiMenu = ({
 
   useEffect(() => {
     if (apiKey) {
-      setApiFree(false);
+      setSelectedApiFree(false);
```

Comment:
The original intention of the following code is to automatically switch the user to use their own api key if they already have one stored inside their `localstorage`. In addition, the code would automatically toggle the radio button to the custom api option when they open up the menu.
```javascript
useEffect(() => {
    if (apiKey) {
      setApiFree(false);
      _setApiKey(apiKey);
    }
  }, []);
```

I would suggest that we keep `setApiFree(false)` and add another line for the local state: `_setApiFree(false)`.